Update shipping partner suggestions to use data from API

ilyasfoo opened the pull request:
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Closes #38456.

Adds consumption of shipping partner suggestions from WCCOM API.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Testing row layout**
1. Go to `WooCommerce > Settings > Advanced > WooCommerce.com` (/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com)
1. Uncheck `Display suggestions within WooCommerce` option and save
1. Go to Settings > General and choose `Italy` country and save
1. Delete all shipping zone in Settings > Shipping and delete the option `woocommerce_admin_created_default_shipping_zones`. 
3. Go to WooCommerce > Home and click on `Add shipping costs`
4. Click on `Save shipping options`
5. Observe SendCloud and Packlink are shown

<img width="762" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/5d149026-afbc-40dd-9b25-198112ad5133">


**Testing column layout**
1. Go to Settings > General and choose `Canada` country and save
1. Delete all shipping zone in Settings > Shipping and delete the option `woocommerce_admin_created_default_shipping_zones`. 
3. Go to WooCommerce > Home and click on `Add shipping costs`
4. Click on `Save shipping options`
5. Observe that ShipStation is shown

<img width="730" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/c829fa35-445b-44bb-8c86-26b539bf687f">


**Optional further testing**
7. Optionally, repeat testing for any countries from this list `'CA', 'AU', 'GB', 'ES', 'IT', 'DE', 'FR', 'MX', 'CO', 'CL', 'AR', 'PE', 'BR', 'UY', 'GT', 'NL', 'AT', 'BE'` and ensure it follows the rules in [DefaultShippingPartners.php](https://github.com/woocommerce/woocommerce/blob/d850244ad63c1db5dd970c3e5eb08c8a5f2ab81e/plugins/woocommerce/src/Admin/Features/ShippingPartnerSuggestions/DefaultShippingPartners.php)

<!-- End testing instructions -->

ilyasfoo committed:
Added shipping partner suggestions data handling in JS

ilyasfoo committed:
Add woocommerce services real info

ilyasfoo committed:
Add dynamic layout components, remove unused JS, added images to proper asset paths

ilyasfoo committed:
Changelog

ilyasfoo commented on plugins/woocommerce/src/Admin/API/PaymentGatewaySuggestions.php:
```diff
@@ -124,7 +124,7 @@ public function get_item_schema() {
 			'$schema'    => 'http://json-schema.org/draft-04/schema#',
 			'title'      => 'payment-gateway-suggestions',
-			'type'       => 'array',
+			'type'       => 'object',
```
Random fix: array in schema was wrong, it's supposed to be an object

ilyasfoo committed:
Update changelog and removed unused variable

ilyasfoo committed:
Remove console.log

ilyasfoo committed:
Lint fix

moon0326 reviewed (commented):
Thank you for working on this 👍 

I have one minor comment regarding the onboarding store changes.

Do you think it's worth implementing shipping method codes in a separate package, similar to [payment gateways](https://github.com/woocommerce/woocommerce/tree/trunk/packages/js/data/src/payment-gateways)? It seems that the logic related to shipping methods does not fit into the onboarding package since it is not used during the onboarding process. However, I also noticed payment suggestion logic included in the onboarding package.


Other than that, everything LGTM 👍

rjchow reviewed (dismissed):
Tests well for Mexico! Wondering why we should set remote suggestions to false to test this though? (tested both true and false and they are both ok)

ilyasfoo committed:
Add shipping methods module in woocommerce/data

ilyasfoo committed:
Update data usage

ilyasfoo committed:
Revert "Added shipping partner suggestions data handling in JS"

This reverts commit 6a87ef2658f8fced639047cbed1cbd8e57e4ffb8.

ilyasfoo commented:
Thanks for the suggestion, @moon0326! I've added the new data module in this commit https://github.com/woocommerce/woocommerce/pull/38457/commits/4ea2c65f47273c320eaa66e62917186ca5504206 and removed the old changes here https://github.com/woocommerce/woocommerce/pull/38457/commits/13386af0494947eaef1264f0897255b9b24ce920.

Please take another look! 🙏 

>Tests well for Mexico! Wondering why we should set remote suggestions to false to test this though? (tested both true and false and they are both ok)

Thanks for testing it out @rjchow! Reasons why:

1. Previously, the old WCCOM data was outdated and would error on some places (now it's [already merged](https://github.com/Automattic/woocommerce.com/pull/17129))
2. Because I'd like to get the offline defaults tested 😄

ilyasfoo committed:
Lint fix

moon0326 commented:
> Thanks for the suggestion, @moon0326! I've added the new data module in this commit https://github.com/woocommerce/woocommerce/commit/4ea2c65f47273c320eaa66e62917186ca5504206 and removed the old changes here https://github.com/woocommerce/woocommerce/commit/13386af0494947eaef1264f0897255b9b24ce920.

LGTM! 👍 🚀
